lib/core/src/main/java/io/atlasmap/core/DefaultAtlasFieldActionService.java

```diff
@@ -547,20 +545,55 @@ public ActionProcessor findActionProcessor(Action action, FieldType sourceType)
             readLock.unlock();
         }
 
-        switch (matches.size()) {
-            case 0:
-                return null;
-            case 1:
-                return matches.get(0);
-            default:
-                if (sourceType != null && !Arrays.asList(FieldType.ANY, FieldType.NONE).contains(sourceType)) {
-                    for (ActionProcessor processor : matches) {
-                        if (sourceType.equals(processor.getActionDetail().getSourceType())) {
-                            return processor;
-                        }
-                    }
+        return findBestActionProcessor(matches, sourceType);
+    }
+
+    public ActionProcessor findActionProcessor(String name, Object value) {
+        FieldType valueType = (value != null ? getConversionService().fieldTypeFromClass(value.getClass()) : FieldType.NONE);
+        String uppercaseName = name.toUpperCase();
+
+        List<ActionProcessor> processors = new ArrayList<>();
+        Lock readLock = actionProcessorsLock.readLock();
+        try {
+            readLock.lock();
+            for (ActionProcessor processor : actionProcessors) {
+                if (processor.getActionDetail().getName().toUpperCase().equals(uppercaseName)) {
+                    processors.add(processor);
                 }
-                return matches.get(0);
+            }
+        } finally {
+            readLock.unlock();
+        }
+
+        return findBestActionProcessor(processors, valueType);
+    }
+
+    private ActionProcessor findBestActionProcessor(List<ActionProcessor> processors, FieldType valueType) {
+        if (processors.isEmpty()) {
+            return null;
+        } else if (processors.size() == 1) {
+            return processors.get(0);
+        } else if (valueType != null && !Arrays.asList(FieldType.ANY, FieldType.NONE).contains(valueType)) {
+            for (ActionProcessor processor: processors) {
+                if (valueType.equals(processor.getActionDetail().getSourceType())) {
+                    return processor;
+                }
+            }
+        }
+        return processors.get(0);
+    }
+
+    public Object processAction(ActionProcessor actionProcessor, Map<String, Object> properties, Object value) {
+        FieldType valueType = (value != null ? getConversionService().fieldTypeFromClass(value.getClass()) : FieldType.NONE);
```

Comment:
Here I determine the type of value as some actions have the same name, but accept different types.